### PR TITLE
Changed code to use integer representations of dates internally.

### PR DESCRIPTION
**Pull request by LyonesGamer**
The code now stores dates as bigint, a signed 64-bit integer, in the database. However, the API still uses ISO format dates.

### CODE REVIEW

**Comment by tschuy** [comment]
Tests, think of the tests! (They're failing.)

File: migrations/20150101000000_00.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ exports.up = function(knex) {
     table.integer('project').references('id').inTable('projects').notNullable();
     table.string('notes');
     table.string('issue_uri');
-    table.timestamp('date_worked');
+    table.bigInteger('date_worked');
     table.timestamps();
```
Comment (tschuy) [inline migrations/20150101000000_00.js]: make sure this makes the right things

File: tests/fixtures/test_data.json [hunk 2]
```diff
@@ -15,6 +15,11 @@
       "name":"Systems",
       "slug":"sys",
       "id": 3
+    },
+    {
+      "name":"Meetings",
+      "slug":"meeting",
+      "id":4
     }
```
Comment (tschuy) [inline tests/fixtures/test_data.json]: this makes me wince

File: tests/times.js [hunk 3]
```diff
@@ -5,26 +5,1287 @@ module.exports = function(expect, request, baseUrl) {
   what should be returned */
   describe('GET /times', function() {
     it('returns all times in the database', function(done) {
-      request.get(baseUrl + 'times', function(err, res, body) {
+      request.get(baseUrl + 'times', function(getErr, getRes, getBody) {
```
Comment (tschuy) [inline tests/times.js]: but not as much as this does

File: migrations/20150101000000_00.js [hunk 4]
```diff
@@ -13,8 +13,9 @@ exports.up = function(knex) {
     table.integer('project').references('id').inTable('projects').notNullable();
     table.string('notes');
     table.string('issue_uri');
-    table.timestamp('date_worked');
-    table.timestamps();
+    table.bigInteger('date_worked').notNullable();
+    table.bigInteger('created_at').notNullable();
+    table.bigInteger('updated_at').defaultTo(-1);
```
Comment (tschuy) [inline migrations/20150101000000_00.js]: why not null?

**Comment by LyonesGamer** [comment]
Rebased to new rebase in parent branch.

File: src/times.js [hunk 5]
```diff
@@ -8,131 +8,206 @@ module.exports = function(app) {
   const passport = require('passport');
 
   app.get(app.get('version') + '/times', function(req, res) {
-    knex('times').then(function(times) {
-      if (times.length === 0) {
-        return res.send([]);
-      }
+    let activitiesList = req.query.activity;
```
Comment (tschuy) [inline src/times.js]: why is this a diff?

File: tests/times.js [hunk 6]
```diff
@@ -5,26 +5,1287 @@ module.exports = function(expect, request, baseUrl) {
...
+
+        expect(getErr).to.equal(null);
+        expect(getRes.statusCode).to.equal(200);
+        expect(JSON.parse(getBody)).to.deep.equal(expectedResults);
+        done();
+      });
+    });
+  });
+
+  describe('GET /times?user=:user', function() {
```
Comment (tschuy) [inline tests/times.js]: why are there new tests in this commit?

File: migrations/20150101000000_00.js (hunk 4 above)
Comment (LyonesGamer) [inline migrations/20150101000000_00.js]: We've always agreed that they shouldn't be null. It just wasn't reflected in the schema.

**Comment by tschuy** [comment]
+1

File: tests/times.js [hunk 7]
```diff
@@ -59,7 +59,7 @@ module.exports = function(expect, request, baseUrl) {
 
         expect(getErr).to.equal(null);
         expect(getRes.statusCode).to.equal(200);
-        expect(JSON.parse(getBody)).to.deep.have.same.members(expectedResults);
+        expect(JSON.parse(getBody)).to.deep.equal(expectedResults);
```
Comment (athai) [inline tests/times.js]: Change ``.to.deep.equal`` back to ``.to.deep.have.same.members`` every time you're comparing the contents of an array. I think it looks good other than that.

**Comment by tschuy** [comment]
+1

**Comment by tschuy** [comment]
+1

**Comment by athai** [comment]
+1

**Pull request merged**